src/h5web/dataset-visualizer/mapper/SlicingSlider.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { ReactElement } from 'react';
+import ReactSlider from 'react-slider';
+import styles from './DimensionMapper.module.css';
+import type { DimensionMapping } from '../models';
+
+interface Props {
+  dimension: number;
+  slicingIndex: number;
+  rawDims: number[];
+  mapperState: NonNullable<DimensionMapping>;
+  onChange: (mapperState: DimensionMapping) => void;
+}
+
+function SlicingSlider(props: Props): ReactElement {
+  const { dimension, slicingIndex, rawDims, mapperState, onChange } = props;
+
+  return (
+    <div key={dimension} className={styles.sliderWrapper}>
+      <span className={styles.sliderLabel}>D{dimension}</span>
+      <ReactSlider
+        key={`${mapperState.includes('y')}`}
+        className={styles.slider}
+        trackClassName={styles.sliderTrack}
+        thumbClassName={styles.sliderThumb}
+        renderThumb={(thumbProps, state) => (
+          <div {...thumbProps}>{state.valueNow}</div>
+        )}
+        value={slicingIndex}
+        onChange={(value) => {
+          const newMapperState = mapperState.slice();
+          newMapperState[dimension] = Number(value);
```

Comment:
Is the `Number(value)` necessary here (rather than `value`) ?

Reply:
You're right, it's more correct to use `as number`. The value is already a number but TS doesn't know it.